static/panes/cfg-view.ts

```diff
@@ -22,427 +22,323 @@
-                {
-                    id: 0,
-                    shape: 'box',
-                    label: '-emit-llvm currently not supported',
-                },
-            ],
-            edges: [],
-        };
-        this.binaryModeSupport = {
-            nodes: [
-                {
-                    id: 0,
-                    shape: 'box',
-                    label: 'Cfg mode cannot be used when the binary filter is set',
-                },
-            ],
-            edges: [],
-        };
+const ColorTable = {
+    red: '#FE5D5D',
+    green: '#76E381',
+    blue: '#65B7F6',
+    grey: '#c5c5c5',
+};
 
-        this.savedPos = state.pos;
-        this.savedScale = state.scale;
-        this.needsMove = this.savedPos && this.savedScale;
+type Coordinate = {
+    x: number;
+    y: number;
+};
 
-        this.currentFunc = state.selectedFn || '';
-        this.functions = {};
+const DZOOM = 0.1;
+const MINZOOM = 0.1;
 
-        this._binaryFilter = false;
-
-        const pickerEl = this.domRoot.find('.function-picker')[0] as HTMLInputElement;
-        this.functionPicker = new TomSelect(pickerEl, {
-            sortField: 'name',
-            valueField: 'name',
-            labelField: 'name',
-            searchField: ['name'],
+export class Cfg extends Pane<CfgState> {
+    graphDiv: HTMLElement;
+    svg: SVGElement;
+    blockContainer: HTMLElement;
+    graphContainer: HTMLElement;
+    graphElement: HTMLElement;
+    infoElement: HTMLElement;
+    currentPosition: Coordinate = {x: 0, y: 0};
+    dragging = false;
+    dragStart: Coordinate = {x: 0, y: 0};
+    dragStartPosition: Coordinate = {x: 0, y: 0};
+    graphDimensions = {width: 0, height: 0};
+    functionSelector: TomSelect;
+    results: CFGResult;
+    state: CfgState & PaneState;
+    layout: GraphLayoutCore;
+    bbMap: Record<string, HTMLDivElement> = {};
+
+    constructor(hub: Hub, container: Container, state: CfgState & PaneState) {
+        if ((state as any).selectedFn) {
+            state = {
+                id: state.id,
+                compilerName: state.compilerName,
+                editorid: state.editorid,
+                treeid: state.treeid,
+                selectedFunction: (state as any).selectedFn,
+                zoom: 1,
+            };
+        }
+        super(hub, container, state);
+        this.eventHub.emit('cfgViewOpened', this.compilerInfo.compilerId);
+        this.eventHub.emit('requestFilters', this.compilerInfo.compilerId);
+        this.eventHub.emit('requestCompiler', this.compilerInfo.compilerId);
+        const selector = this.domRoot.get()[0].getElementsByClassName('function-selector')[0];
+        if (!(selector instanceof HTMLSelectElement)) {
+            throw new Error('.function-selector is not an HTMLSelectElement');
+        }
+        this.functionSelector = new TomSelect(selector, {
+            valueField: 'value',
+            labelField: 'title',
+            searchField: ['title'],
```

Comment:
I think adding `placeholder` here like in compiler-picker (https://github.com/compiler-explorer/compiler-explorer/blob/main/static/compiler-picker.ts#L98) would help convey that you can type to search

Reply:
Great idea and I also added `plugins: ['dropdown_input']` like the compiler picker so the search line is less crowded